data-structures/linked-lists-1.ts

```diff
@@ -0,0 +1,21 @@
+/* eslint-disable @typescript-eslint/no-empty-function */
+import { LinkedList } from './lib/linked-list';
+
+export function getHead<T>(list: LinkedList<T>): T | null {
+  return list.data;
+}
+
+export function updateHead<T>(list: LinkedList<T>, value: T | null): void {
+  list.data = value;
+}
+
+export function withoutHead<T>(list: LinkedList<T>): LinkedList<T> | null {
+  return list.next;
+}
+
+export function prepend<T>(list: LinkedList<T>, value: T): LinkedList<T> {
+  const node = new LinkedList(value);
+  node.next = list;
+  list = node;
```

Comment:
Changes to `list` do not get seen by the caller, so this assignment isn't necessary. You can just `return node;`.